src/ws/service/friendships_service.rs

```diff
@@ -67,21 +72,24 @@ impl FriendshipsServiceServer<SocialContext, RPCFriendshipsServiceError> for MyF
         request: Payload,
         context: ProcedureContext<SocialContext>,
     ) -> Result<ServerStreamResponse<UsersResponse>, RPCFriendshipsServiceError> {
+        let start_time = Instant::now();
+        let metrics = context.server_context.metrics.clone();
+        record_in_procedure_call_size(metrics.clone(), Procedure::GetFriends, &request).await;
+
         let request_user_id = get_user_id_from_request(
             &request,
             context.server_context.synapse.clone(),
             context.server_context.users_cache.clone(),
         )
         .await;
 
-        let metrics = context.server_context.metrics.clone();
-
         let (friendships_generator, friendships_yielder) = Generator::create();
 
         let Some(repos) = context.server_context.db.db_repos.clone() else {
             log::error!("[RPC] Get friends > Db repositories > `repos` is None.");
             let error = InternalServerError{ message: "An error occurred while getting the friendships".to_owned() };
-            record_procedure_call(metrics,Some(error.clone().into()), Procedure::GetFriends).await;
+            record_procedure_call_and_duration_and_size(metrics.clone(), Some(error.clone().into()), Procedure::GetFriends, start_time, error.encoded_len()).await;
```

Comment:
what about renaming this to `out`?
```suggestion
            record_procedure_call_and_duration_and_size(metrics.clone(), Some(error.clone().into()), Procedure::GetFriends, start_time, error.encoded_len()).await;
```